File: skills/astra-toolkit/clients/go/examples/tables-create-index-exists.go
package main

import (
	"context"
	"log"
	"os"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get an existing table
	client := astra.NewClient()

	database := client.Database(
		os.Getenv("ASTRA_DB_API_ENDPOINT"),
		options.API().SetToken(os.Getenv("ASTRA_DB_APPLICATION_TOKEN")),
	)

	table := database.Table("**TABLE_NAME**")

	// Index a column
	err := table.CreateIndex(
		ctx,
		"**INDEX_NAME**",
		"**COLUMN_NAME**",
		options.CreateIndex().SetIfNotExists(true),
	)
	if err != nil {
		log.Fatal(err)
	}
}

File: skills/astra-toolkit/clients/csharp/examples/tables-create-index-exists.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Tables;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      System.Environment.GetEnvironmentVariable("ASTRA_DB_API_ENDPOINT"),
      System.Environment.GetEnvironmentVariable("ASTRA_DB_APPLICATION_TOKEN")
    );
    var table = database.GetTable("**TABLE_NAME**");

    // Index a column
    await table.CreateIndexAsync(
      "**INDEX_NAME**",
      "**COLUMN_NAME**",
      new CreateIndexOptions() { IfNotExists = true }
    );
  }
}
